### PR TITLE
fix(web): add onData prop to MultiList

### LINKED ISSUE #1603: Items Count next to MutliList Title
**Affected Projects**

React

**Describe the solution you'd like**

Is it possible to show the count of number of items in a MultiList? For example, the max size is set to 100, but the list could have 50 elements. Can we show an Item count next to the Title? 

**Additional context**
image

### PR DESCRIPTION
**Pull request by vedadeepta**
**Before submitting a pull request,** please make sure the following is done:

- [x] Describe the proposed changes and how it'll improve the library experience.
Fix #1603 
- [x] Please make sure that there are no linting errors in the code.
- [x] Add a demo video/gif/screenshot to explain how did you test the fix.
appbase-bug

- [ ] If it is a global change, try to add any side effects that it could have.
- [x] Create a PR to add/update the docs (if needed) at [here](https://github.com/appbaseio/Docs).
https://github.com/appbaseio/Docs/pull/179/files
- [x] Create a PR to add/update the storybook (if needed) at [here](https://github.com/appbaseio/playground).
https://github.com/appbaseio/playground/pull/64

### CODE REVIEW

File: packages/web/src/components/list/MultiList.js [hunk 1]
```diff
@@ -151,6 +151,21 @@ class MultiList extends Component {
 			this.updateQuery(valueArray, this.props);
 		});
 
+		// checkPropChange(this.props.options, prevProps.options, () => {
```
Comment (bietkul) [inline packages/web/src/components/list/MultiList.js]: @vedadeepta Remove comments

**Review by bietkul — CHANGES_REQUESTED** [review]
@vedadeepta Thanks for the PR. I would suggest the following changes to include this feature in the next release.
- [ ] `onData` callback should provide the following properties https://github.com/appbaseio/reactivesearch/blob/next/packages/web/src/components/list/MultiList.js#L495. You can define a getter method and use it at both places (custom render method and onData callback). I would recommend checking the logic in the `ReactiveList` component.
- [ ] It should get invoked whenever any of the following dependencies changed.
`error`, `isLoading`, `options` and `selectedValue`
- [ ] Can you define `onData` callback in all the list components?
`SingleList`, `MultiList`, `SingleDropdownList`, `MultiDropdownList`, `SingleDataList` and `MultiDataList`
- [ ] Update the docs and playground to cover the other list components

**Comment by vedadeepta** [comment]
@bietkul both the local and prod environment errors out on switching stories (someitmes)

https://opensource.appbase.io/playground/?path=/story/list-components-multilist--basic

try switching stories. 

Error on prod:

>An error has occured. You're using Reactivesearch Version: 3.16.2. If you think this is a problem with Reactivesearch, please try updating to the latest version. If you're already at the latest version, please open an issue at https://github.com/appbaseio/reactivesearch/issues Error: Minified React error #31; visit https://reactjs.org/docs/error-decoder.html?invariant=31&args[]=object%20with%20keys%20%7Btook%7D&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at Qg (vendors~main.a241fb21564d84851cb3.bundle.js:1)
    at p (vendors~main.a241fb21564d84851cb3.bundle.js:1)
    at ca (vendors~main.a241fb21564d84851cb3.bundle.js:1)
    at vendors~main.a241fb21564d84851cb3.bundle.js:1
    at R (vendors~main.a241fb21564d84851cb3.bundle.js:1)
    at Rj (vendors~main.a241fb21564d84851cb3.bundle.js:1)
    at Qj (vendors~main.a241fb21564d84851cb3.bundle.js:1)
    at Kj (vendors~main.a241fb21564d84851cb3.bundle.js:1)
    at yj (vendors~main.a241fb21564d84851cb3.bundle.js:1)
    at vendors~main.a241fb21564d84851cb3.bundle.js:1 {componentStack: "↵    in p↵    in ForwardRef(render)↵    in div↵   …ReadmeContent↵    in storyFn↵    in ErrorBoundary"}
    
   Error on local env
   
   >An error has occured. You're using Reactivesearch Version: 3.16.2. If you think this is a problem with Reactivesearch, please try updating to the latest version. If you're already at the latest version, please open an issue at https://github.com/appbaseio/reactivesearch/issues Error: Objects are not valid as a React child (found: object with keys {took}). If you meant to render a collection of children, use an array instead.
    in p (created by Context.Consumer)
    in EmotionCssPropInternal (created by ReactiveList)
    in div (created by Context.Consumer)
    in Flex (created by ReactiveList)
    in div (created by ReactiveList)
    in ReactiveList (created by ComponentWrapper)
    in ComponentWrapper (created by Context.Consumer)
    in Connect(ComponentWrapper)
    in Unknown (created by Context.Consumer)
    in WithTheme(Component) (created by Context.Consumer)
    in Connect(WithTheme(Component)) (created by ForwardRef)
    in ForwardRef (created by MultiListDefault)
    in div (created by MultiListDefault)
    in div (created by MultiListDefault)
    in div
    in Unknown (created by Context.Consumer)
    in Base (created by URLParamsProvider)
    in URLParamsProvider
    in Unknown (created by Context.Consumer)
    in Connect(Component) (created by ForwardRef)
    in ForwardRef (created by ReactiveBase)
    in Provider (created by ReactiveBase)
    in ThemeProvider (created by ReactiveBase)
    in ReactiveBase (created by MultiListDefault)
    in MultiListDefault (created by storyFn)
    in div (created by _default)
    in div (created by _default)
    in _default (created by ReadmeContent)
    in div (created by ReadmeContent)
    in ReadmeContent (created by storyFn)
    in storyFn
    in ErrorBoundary
    at throwOnInvalidObjectType (react-dom.development.js:13413)
    at createChild (react-dom.development.js:13651)
    at reconcileChildrenArray (react-dom.development.js:13900)
    at reconcileChildFibers (react-dom.development.js:14305)
    at reconcileChildren (react-dom.development.js:16762)
    at updateHostComponent (react-dom.development.js:17302)
    at beginWork (react-dom.development.js:18627)
    at HTMLUnknownElement.callCallback (react-dom.development.js:188)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:237)
    at invokeGuardedCallback (react-dom.development.js:292) {componentStack: "↵    in p (created by Context.Consumer)↵    in Emo…d by storyFn)↵    in storyFn↵    in ErrorBoundary"}

**Comment by bietkul** [comment]
@vedadeepta We have fixed this issue.

**Pull request closed without merging**